livekit-agents/livekit/agents/llm/chat_context.py

```diff
@@ -27,22 +27,24 @@ class ChatRole(enum.Enum):
     TOOL = "tool"
 
 
+@dataclass
+class ChatImage:
+    image: str | rtc.VideoFrame
+    inference_width: int = 128
+    inference_height: int = 128
+    _cache: Dict[object, Any] = field(default_factory=dict)
```

Comment:
Not fan of having private variables in a dataclass, but it is OK for now, we can iterate with a new ChatContext API.
For now let's correctly hide the field
```suggestion
    _cache: dict = field(default_factory=dict, init=False, repr=False)
```

Reply:
I see we're not using a LLM base class for the keys. I guess the reason is to avoid import cycles. 
Not sure if moving this stuff inside a separate file is really necessary?

Reply:
Yeah I guess since it's only for a type I can circular import with TYPE_CHECKING check. I think the separate file has nice symmetry with function_context.py and since this is potentially still going to grow I think it makes sense.